src/identity-provider.js

```diff
@@ -1,57 +1,23 @@
 'use strict'
-const isDefined = require('./is-defined')
-const Identity = require('./identity')
-const defaultType = 'orbitdb'
-
 class IdentityProvider {
-  constructor (keystore) {
-    if (!isDefined(keystore)) {
-      throw new Error('Keystore is required')
-    }
-    this._keystore = keystore
-  }
-
-  async createIdentity(id, options = {}) {
-    // Get the key for id from the keystore or create one
-    // if it doesn't exist
-    const key = await this._keystore.getKey(id) ||
-      await this._keystore.createKey(id)
-    // Sign with the key for the id
-    const selfSigningFn = async (id, data) => await this._keystore.sign(key, data)
-    // If no type was indicated, set to default
-    const type = isDefined(options.type) ? options.type : defaultType
-    // If signing function was not passed, use keystore as the identity signer
-    const identitySignerFn = isDefined(options.identitySignerFn) ? options.identitySignerFn : selfSigningFn
-    // Sign the id with the signing key we're going to use
-    const idSignature = await this._keystore.sign(key, id)
-    // Get the hex string of the public key
-    const publicKey = key.getPublic('hex')
-    // Sign both the key and the signature created with that key
-    const pubKeyIdSignature = await identitySignerFn(id, publicKey + idSignature)
-    return new Identity(id, publicKey, idSignature, pubKeyIdSignature, type, this)
-  }
+  constructor() {}
 
-  static async createIdentity (keystore, id, options = {}) {
-    const identityProvider = new IdentityProvider(keystore)
-    return await identityProvider.createIdentity(id, options)
-  }
-
-  static async verifyIdentity (identity, verifierFunction) {
-    return verifierFunction(identity)
-  }
+  /* Return publicKey of identity */
+  async createId(options) {}
 
-  async sign (identity, data) {
-    const signingKey = await this._keystore.getKey(identity.id)
+  /* Return signature of OrbitDB public key signature */
+  async signIdentity(pubKeySignature, identity) {}
 
-    if (!signingKey)
-      throw new Error(`Private signing key not found from Keystore`)
+  /* Verify a signature of OrbitDB public key signature */
+  static async verifyIdentity (identity, identityProvider, options) {}
```

Comment:
The verifyIdentity function has a different signature here than in https://github.com/orbitdb/orbit-db-identity-provider/pull/13/files#diff-e17a429b402bcd7bc701a8c9caab648eR32 and https://github.com/orbitdb/orbit-db-identity-provider/pull/13/files#diff-c172505c0fd2b8d78019fb7c97cf1e56R36. These should all match (and if we remove the possibility to pass in a custom keystore for verifyIdentity and signIdentity, we can remove the options from the args, too). Note also that `identity` arg contains `identityProvider`, so that shouldn't be needed either.

Reply:
This is true but the identities which need to be verified don't contain provider (the identities are a result of identity.toJSON() and come from peers so dont contain provider field), though the log does...the issue is that odb-keystore requires an instance of keystore to verify signatures which shouldn't be necessary. Would it cause too much bloating to make odb-keystore.verify static? https://github.com/orbitdb/orbit-db-keystore/blob/0f19bde078148a25b6d4e331c9a6986a10691080/src/keystore.js#L80